fcrepo-indexing-triplestore/src/main/java/org/fcrepo/camel/indexing/triplestore/TriplestoreRouter.java

```diff
@@ -98,47 +100,65 @@ public void configure() throws Exception {
+            .choice()
+            .when(simple(config.isIndexingPredicate() + " != 'true'"))
+            .to("direct:update.triplestore")
+            .otherwise()
+            .to("fcrepo:" + config.getFcrepoBaseUrl() +
+                "?preferInclude=PreferMinimalContainer&accept=application/rdf+xml")
+            .choice()
+            .when(indexable)
+            .to("direct:update.triplestore")
+            .otherwise()
+            .to("direct:delete.triplestore");
 
         /**
          * Remove an item from the triplestore index.
          */
         from("direct:delete.triplestore")
-                .routeId("FcrepoTriplestoreDeleter")
-                .process(new SparqlDeleteProcessor())
-                .log(LoggingLevel.INFO, LOGGER,
-                        "Deleting Triplestore Object ${headers[CamelFcrepoUri]}")
-                .to(config.getTriplestoreBaseUrl() + "?useSystemProperties=true");
+            .routeId("FcrepoTriplestoreDeleter")
+            .process(new SparqlDeleteProcessor())
+            .log(LoggingLevel.INFO, LOGGER,
+                "Deleting Triplestore Object ${headers[CamelFcrepoUri]}")
+            .choice()
+            .when((x) -> !config.getTriplestoreAuthUsername().isEmpty())
+            .setHeader("Authorization", simple(
+                "Basic " + Base64.getEncoder().encodeToString(
+                    (config.getTriplestoreAuthUsername() +
+                        ":" +
+                        config.getTriplestoreAuthPassword()
+                    ).getBytes()))
+            )
+            .end()
+            .to(config.getTriplestoreBaseUrl() + "?useSystemProperties=true");
 
         /**
          * Perform the sparql update.
          */
         from("direct:update.triplestore")
-                .routeId("FcrepoTriplestoreUpdater")
-                .setHeader(FCREPO_NAMED_GRAPH)
-                .simple(config.getNamedGraph())
-                .to("fcrepo:" + config.getFcrepoBaseUrl() + "?accept=application/n-triples" +
-                        "&preferOmit=" + config.getPreferOmit() + "&preferInclude=" + config.getPreferInclude())
-                .process(new SparqlUpdateProcessor())
-                .log(LoggingLevel.INFO, LOGGER,
-                        "Indexing Triplestore Object ${headers[CamelFcrepoUri]}")
-                .to(config.getTriplestoreBaseUrl() + "?useSystemProperties=true");
+            .routeId("FcrepoTriplestoreUpdater")
+            .setHeader(FCREPO_NAMED_GRAPH)
+            .simple(config.getNamedGraph())
+            .to("fcrepo:" + config.getFcrepoBaseUrl() + "?accept=application/n-triples" +
+                "&preferOmit=" + config.getPreferOmit() + "&preferInclude=" + config.getPreferInclude())
+            .process(new SparqlUpdateProcessor())
+            .log(LoggingLevel.INFO, LOGGER,
+                "Indexing Triplestore Object ${headers[CamelFcrepoUri]}")
+            .setHeader("Authorization", simple(
```

Comment:
I think we need the conditional here for choosing when to add the Authorization header. I was getting a 401 without it when doing a POST to create a new resource.